pq/stateful_hash_sig/lms_sign_verify.c

```diff
@@ -0,0 +1,396 @@
+    fprintf(stderr, "  number of signatures (1 by default).\n");
+    fprintf(stderr, "\n");
+    fprintf(stderr, "  If 0 is given for num signatures, it prints the private and\n");
+    fprintf(stderr, "  public key as hex and exits early.\n");
+    fprintf(stderr, "\n");
+    fprintf(stderr, "notes:\n");
+    fprintf(stderr, " - The acceptable parameter values are those in RFC8554:\n");
+    fprintf(stderr, "     levels = {1..8}\n");
+    fprintf(stderr, "     height = {5, 10, 15, 20, 25}\n");
+    fprintf(stderr, "     Winternitz = {1, 2, 4, 8}\n");
+    fprintf(stderr, "\n");
+    fprintf(stderr, " - The number of available signatures is\n");
+    fprintf(stderr, "     n = 2 ** (levels * height)\n");
+    fprintf(stderr, "\n");
+    fprintf(stderr, " - The Signature size is directly proportional to the levels value,\n");
+    fprintf(stderr, "   and inversely proportional to the Winternitz value. However it\n");
+    fprintf(stderr, "   grows only modestly with the height parameter.\n");
+    fprintf(stderr, "\n");
+    fprintf(stderr, " - The Winternitz value is a space-time tradeoff parameter. Larger values\n");
+    fprintf(stderr, "   will reduce signature size, while increasing times for key generation,\n");
+    fprintf(stderr, "   signing, and verifying.\n");
+    fprintf(stderr, "\n");
+    fprintf(stderr, " - Key generation time is strongly determined by the height of\n");
+    fprintf(stderr, "   the first level tree. A 3 level, 5 height tree is much faster\n");
+    fprintf(stderr, "   than 1 level, 15 height at initial key gen, even if the number\n");
+    fprintf(stderr, "   of available signatures is the same.\n");
+
+    exit(EXIT_FAILURE);
+}
+
+static int
+write_key_file(const byte * priv,
+               word32       privSz,
+               void *       context)
+{
+    FILE *       file = NULL;
+    const char * filename = NULL;
+    int          n_cmp = 0;
+    size_t       n_read = 0;
+    size_t       n_write = 0;
+    byte         buff[HSS_MAX_PRIVATE_KEY_LEN];
+    int          err = 0;
+
+    if (priv == NULL || context == NULL || privSz == 0) {
+        fprintf(stderr, "error: invalid write args\n");
+        return WC_LMS_RC_BAD_ARG;
+    }
+
+    filename = context;
+
+    /* Open file for read and write. */
+    file = fopen(filename, "r+");
+    if (!file) {
+        /* Create the file if it didn't exist. */
+        file = fopen(filename, "w+");
+        if (!file) {
+            fprintf(stderr, "error: fopen(%s, \"w+\") failed: %d\n", filename,
+                    ferror(file));
+            return WC_LMS_RC_WRITE_FAIL;
+        }
+    }
+
+    n_write = fwrite(priv, 1, privSz, file);
+
+    if (n_write != privSz) {
+        fprintf(stderr, "error: wrote %zu, expected %d: %d\n", n_write, privSz,
+                ferror(file));
+        return WC_LMS_RC_WRITE_FAIL;
+    }
+
+    rewind(file);
+
+    XMEMSET(buff, 0, n_write);
+
```

Comment:
Add comment here that you are checking that the state has been properly written to the disk.

Reply:
done!